File: baekjoon/1260.py
from collections import deque

N, M, V = list(map(int, input().split()))
graph = [[] for _ in range(N + 1)]
for _ in range(M):
    start, end = list(map(int, input().split()))
    graph[start].append(end)
    graph[end].append(start)

# DFS
def dfs(start, visited = []):
    visited.append(start)
    for next in sorted(graph[start]):
        if next not in visited:
            dfs(next, visited)
    return visited

# BFS
def bfs(start):
    visited = [start]
    q = deque([start])

    while q:
        node = q.popleft()
        for next in sorted(graph[node]):
            if next not in visited:
                visited.append(next)
                q.append(next)
    return visited

# 리스트 출력 형식 변환
def changeStr(numlist):
    return ' '.join(list(map(str, numlist)))

print(changeStr(dfs(V)), changeStr(bfs(V)), sep='\n')

File: baekjoon/1260.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt';
const [L, ...datas] = fs.readFileSync(filePath).toString().trim().split('\n');

const solution = (N, M, V, list) => {
  const graph = Array.from({ length: N + 1 }, () => []);
  list.forEach((pair) => {
    graph[pair[0]].push(pair[1]);
    graph[pair[1]].push(pair[0]);
  });

  // DFS
  const dfs = (start, visited = []) => {
    visited.push(start);
    graph[start]
      .sort((a, b) => a - b)
      .forEach((next) => {
        if (!visited.includes(next)) {
          dfs(next, visited);
        }
      });
    return visited;
  };

  // BFS
  const bfs = (start) => {
    const visited = [start];
    const q = [start];

    while (!!q.length) {
      let node = q.shift();
      graph[node]
        .sort((a, b) => a - b)
        .forEach((next) => {
          if (!visited.includes(next)) {
            visited.push(next);
            q.push(next);
          }
        });
    }
    return visited;
  };
  console.log([dfs(V).join(' '), bfs(V).join(' ')].join('\n'));
};

solution(
  ...L.split(' ').map(Number),
  datas.map((x) => x.split(' ').map(Number))
);
